components/script/dom/serviceworkerjob.rs

```diff
@@ -22,27 +25,40 @@ pub enum JobType {
     Update
 }
 
+#[derive(Clone)]
+pub enum SettleType {
+    Resolve(Trusted<ServiceWorkerRegistration>),
+    Reject(Error)
+}
+
 #[must_root]
 pub struct Job {
     pub job_type: JobType,
     pub scope_url: Url,
     pub script_url: Url,
     pub promise: Rc<Promise>,
     pub equivalent_jobs: Vec<Job>,
+    // client can be a window client, worker client so `Client` will be an enum in future
+    pub client: Trusted<Client>,
```

Comment:
It's odd to use Trusted and not TrustedPromise here. Why can't we use `JS<Client>`?